docs/REST API/rest-api.md

```diff
@@ -6932,12 +6938,104 @@ Deletes the session specified by ID. When the user associated with the session n
+- [Get software title](#get-software-title)
+- [Get software version](#get-software-version)
+
+### List software titles
+
+Get a list of all software grouped by software title.
+
+`GET /api/v1/fleet/software/titles`
+
+#### Parameters
+
+| Name                    | Type    | In    | Description                                                                                                                                                                |
+| ----------------------- | ------- | ----- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| page                    | integer | query | Page number of the results to fetch.                                                                                                                                       |
+| per_page                | integer | query | Results per page.                                                                                                                                                          |
+| order_key               | string  | query | What to order results by. Allowed fields are `name`, `hosts_count`, `vulnerabilities_count`. Default is `hosts_count` (descending).      |
+| order_direction         | string  | query | **Requires `order_key`**. The direction of the order given the order key. Options include `asc` and `desc`. Default is `asc`.                                              |
+| query                   | string  | query | Search query keywords. Searchable fields include `title` and `cve`.                                                                                             |
+| team_id                 | integer | query | _Available in Fleet Premium_ Filters the software to only include the software installed on the hosts that are assigned to the specified team.                             |
+| vulnerable              | bool    | query | If true or 1, only list software that has detected vulnerabilities. Default is `false`.                                                                                    |
+
+#### Example
+
+`GET /api/v1/fleet/software/titles`
+
+##### Default response
+
+`Status: 200`
+
+```json
+{
+  "counts_updated_at": "2022-01-01 12:32:00",
+  "count": 2,
+  "software_titles": [
+    {
+      "id": 12,
+      "name": "Firefox.app",
+      "versions_count": 3,
+      "source": "apps",
+      "hosts_count": 48,
+      "versions": [ 
+        {
+          "id": 123,
+          "version": "1.12",
+          "vulnerabilities": ["CVE-2023-1234","CVE-2023-4321","CVE-2023-7654"]
+        }, 
+        {
+          "id": 124, 
+          "version": "3.4",
+          "vulnerabilities": ["CVE-2023-1234","CVE-2023-4321","CVE-2023-7654"]
+        },
+        {
+          "id": 12
+          "version": "1.13",
+          "vulnerabilities": ["CVE-2023-1234","CVE-2023-4321","CVE-2023-7654"]
```

Comment:
@noahtalerman @marko-lisica what if the software doesn't have any vulnerabilities? empty array or `null`? cc: @ghernandez345

Reply:
no reason, just sanity checking. Thanks!